src/Core/BasePlugin.php

```diff
@@ -315,9 +315,10 @@ public function services(ContainerInterface $container): void
      * Register application events.
      *
      * @param \Cake\Event\EventManagerInterface $eventManager The global event manager to register listeners on
+     * @param \Cake\Core\ContainerInterface $container The DI container instance
      * @return \Cake\Event\EventManagerInterface
      */
-    public function events(EventManagerInterface $eventManager): EventManagerInterface
+    public function events(EventManagerInterface $eventManager, ContainerInterface $container): EventManagerInterface
```

Comment:
This will break every plugin which has used this events hook.

Reply:
I thought since 5.1.0 release is < than 1 month ago these kind of changes would be fine to do since not many users will have adopted this kind of change.

Technically of course this is a breaking change, but even my alternative of injecting the container into the plugin constructor can't be done safely in 5.next either.

So should this really only be shipped to users in 6.x in ~3 years?

Reply:
I have an idea...